include/zephyr/drivers/gpio.h

```diff
@@ -428,6 +428,7 @@ struct gpio_dt_spec {
  * in the device structure.
  */
 struct gpio_driver_config {
+	DEVICE_MMIO_ROM; /* Must be first */
```

Comment:
This should be in a separate commit with big fat explanation why this is the correct change.